### DISCUSSION TITLE
How to detect request headers stored in the library

### DISCUSSION THREAD

**Discussion by Evilglory1371** [corazawaf/coraza#931] [Q&A]
Hello, I have a question. What should I do to detect known request headers stored in the database? How to operate it specifically, for example, when I retrieve a request header from the database, where should I put it for detection?

**Comment by M4tteoP** [comment]
Hey, you could write a custom connector that reads information from the database and populates a Coraza transaction with all the data that you wish to analyze.
To give you an idea, you can take a look at tests like: https://github.com/corazawaf/coraza/blob/f1cfd138d37a320463e02c831520aec9d73bc75d/internal/corazawaf/transaction_test.go#L199-L213.
Here rules are added to the WAF instance, a transaction is created, and request headers are added and processed. Finally, it is checked if an interruption occurred.
